presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/TopNMatcher.java

```diff
@@ -61,18 +59,12 @@ public MatchResult detailMatches(PlanNode node, StatsProvider stats, Session ses
             return NO_MATCH;
         }
 
-        List<Symbol> expectedOrderBy = orderBySymbols.stream()
-                .map(alias -> alias.toSymbol(symbolAliases))
-                .collect(toImmutableList());
-
-        if (!topNNode.getOrderingScheme().getOrderBy().equals(expectedOrderBy)) {
-            return NO_MATCH;
-        }
-
-        Map<Symbol, SortOrder> expectedOrderings = Maps.toMap(expectedOrderBy, Functions.constant(SortOrder.ASC_NULLS_FIRST));
-
-        if (!topNNode.getOrderingScheme().getOrderings().equals(expectedOrderings)) {
-            return NO_MATCH;
+        for (Ordering ordering : orderBy) {
+            Symbol symbol = Symbol.from(symbolAliases.get(ordering.getField()));
+            OrderingScheme nodeOrderingScheme = topNNode.getOrderingScheme();
```

Comment:
I think you missed the part where you check that order of ordering symbols is correct